src/Lykke.Service.BlockchainCashoutPreconditionsCheck.Services/BlockchainApiClientProvider.cs

```diff
@@ -1,17 +1,23 @@
-﻿using Autofac.Features.Indexed;
+﻿using System;
 using Lykke.Service.BlockchainApi.Client;
 using Lykke.Service.BlockchainCashoutPreconditionsCheck.Core.Exceptions;
 using Lykke.Service.BlockchainCashoutPreconditionsCheck.Core.Services;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Lykke.Common.Log;
+using Lykke.Service.BlockchainCashoutPreconditionsCheck.Core.Settings.ServiceSettings;
 
 namespace Lykke.Service.BlockchainCashoutPreconditionsCheck.Services
 {
     public class BlockchainApiClientProvider : IBlockchainApiClientProvider
     {
-        private readonly IIndex<string, IBlockchainApiClient> _clients;
+        private readonly IImmutableDictionary<string, IBlockchainApiClient> _clients;
 
-        public BlockchainApiClientProvider(IIndex<string, IBlockchainApiClient> clients)
+        public BlockchainApiClientProvider(BlockchainsIntegrationSettings settings, ILogFactory logFactory, int blockchainApiTimeoutSeconds)
         {
-            _clients = clients;
+            var timeout = TimeSpan.FromSeconds(blockchainApiTimeoutSeconds);
+            _clients = settings?.Blockchains.ToImmutableDictionary(x => x.Type, 
```

Comment:
Does _clients field may be null? Later it's used without checks.

Reply:
fixed